main.go

```diff
@@ -59,6 +59,7 @@ func init() {
 	// +kubebuilder:scaffold:scheme
 }
 
+// nolint:gocyclo
```

Comment:
@ncdc temporarily adding this to let tests proceed. Will wait for others to comment on the approach before moving forward.

Reply:
This is probably ok to keep, given that the function is generated

Reply:
It's fine for now. We can fix in a separate PR.